### PR TITLE
Add `stitched_vamana` code

### PR DESCRIPTION
**Pull request by darvg**
- [ ] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [x] Does this PR modify any existing APIs?
   - [x] Is the change to the API backwards compatible?
- [ ] Should this result in any changes to our documentation, either updating existing docs or adding new ones?
 
#### What does this implement/fix? Briefly explain your changes.
Introduces the `stitched_vamana` algorithm as described in the WWW paper. Key changes include a new test file for building such an index, and a small backwards-compatible API addition to `src/index.cpp`.

#### Any other comments?
While the commit ordering is correct, the timestamps of the commits are off due to me cherry-picking them from an old branch.

### CODE REVIEW

File: tests/build_stitched_index.cpp [hunk 1]
```diff
@@ -0,0 +1,811 @@
+// author: Siddharth Gollapudi
+// email: t-gollapudis@microsoft.com
+
+#include <boost/program_options.hpp>
+#include <bits/types/struct_iovec.h>
```
Comment (harsha-simhadri) [inline tests/build_stitched_index.cpp:5]: Can we replace this std library equivalents? Windows build fails here.

**Review by harsha-simhadri — CHANGES_REQUESTED** [review]
Can we replace struc_iovec.h with std library equivalents? Windows build fails here.

File: tests/build_stitched_index.cpp (hunk 1 above)
Comment (darvg) [reply tests/build_stitched_index.cpp:5]: Neel and I did some work on designing an STL algorithm, and they all had their downsides. I would like to keep this implementation due to performance reasons when running on Linux, but I can include an alternative algorithm for Windows platforms?

**Comment by harsha-simhadri** [comment]
@darvg Can you please close this PR and raise a new one to merge to 'dkaczynski/filter' branch. From there on, your changes can get in to main
We are no longer trying to merge ravi/filter_support_rebased to main.

**Comment by harsha-simhadri** [comment]
@darvg Can you please close this PR and raise a new one to merge to 'dkaczynski/filter' branch. From there on, your changes can get in to main
We are no longer trying to merge ravi/filter_support_rebased to main.

**Pull request closed without merging**